### PR TITLE
Auto shoot command group

### PR DESCRIPTION
**Pull request by EliyaFishman**

### CODE REVIEW

File: src/main/java/frc/robot/commands/command_groups/AutoShoot.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.function.DoubleSupplier;
+import java.util.function.Supplier;
+
+import static frc.robot.Robot.drivetrain;
+import static frc.robot.Robot.shooter;
+
+/**
+ * This command group responsible for spinning the mixer, rotating the drivetrain to target, spinning the shooter in the desired speed,
+ * and loading and shooting the cells once it reaches the desired speed.
+ *
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/commands/command_groups/AutoShoot.java]: blank

File: src/main/java/frc/robot/commands/command_groups/AutoShoot.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+	/**
+	 * @param speedSupplier supplier of the desired speed in RPM
+	 */
+	public AutoShoot(DoubleSupplier speedSupplier) {
+		CheesySetShooterVelocity setShooterVelocity = new CheesySetShooterVelocity(speedSupplier);
+		addCommands(
+				setShooterVelocity,
+				new SpinMixer(),
+				sequence(
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/commands/command_groups/AutoShoot.java]: space order

File: src/main/java/frc/robot/constants/RobotConstants.java [hunk 3]
```diff
@@ -72,6 +73,7 @@
         public double THRESHOLD_LIMIT;
         public double TIMEOUT;
         public double TICKS_PER_ROTATION;
+        public double kDefaultSpeed;
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/constants/RobotConstants.java]: velocity

File: src/main/java/frc/robot/subsystems/mixer/SpinMixer.java [hunk 4]
```diff
@@ -32,7 +39,7 @@ public SpinMixer(DoubleSupplier power) {
     this.power = power;
   }
 
-  @Override
+	@Override
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/mixer/SpinMixer.java]: why space?

File: src/main/java/frc/robot/subsystems/shooter/CheesySetShooterVelocity.java [hunk 5]
```diff
@@ -166,4 +165,14 @@ public boolean isFinished() {
...
+
+	/**
+	 * @return whether the shooter is on target velocity.
+	 * Tolerance is taken from {@link frc.robot.constants.RobotConstants.ControlConstants#leftShooterSettings}
+	 */
+	public boolean isOnTarget() {
+		return Math.abs(shooter.getAverageSpeed() - setpoint) <
+				robotConstants.controlConstants.leftShooterSettings.getTolerance();
+	}
+
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/shooter/CheesySetShooterVelocity.java:177]: space

File: src/main/java/frc/robot/utils/DashboardDataContainer.java [hunk 6]
```diff
@@ -35,6 +36,10 @@ public DashboardDataContainer() {
 				SmartDashboard.getNumber("Shooter/Shooting Velocity Setpoint", 0)));
 		SmartDashboard.putData("Shooter/Enable Tuning",
 				new StartEndCommand(shooter::enableTuning, shooter::disableTuning));
+
+		// Command groups data
+		SmartDashboard.putData("CommandGroups/AutoShoot", new AutoShoot(() ->
+				SmartDashboard.getNumber("Shooter/Shooting Velocity Setpoint", 0)));
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/utils/DashboardDataContainer.java]: SmartDashboard.getNumber("Shooter/Shooting Velocity Setpoint", 0)));
add it's putNumber

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
few changes

**Comment by EliyaFishman** [comment]
[Command Group: AutoShoot](https://trello.com/c/qeH98gCS/107-command-group-autoshoot)

**Review by HilelAmiel — APPROVED** [review]

**Pull request merged**